static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx

```diff
@@ -712,7 +720,11 @@ function WidgetBuilder({
     return false;
   }
 
-  if (isEditing && widgetIndex >= dashboard.widgets.length) {
+  if (
+    isEditing &&
+    (parseInt(widgetIndex, 10) >= dashboard.widgets.length ||
```

Comment:
if we use the parsed widgetIndex in several places, wouldn't it be better if we created a variable for it and only reuse it?

Reply:
Done! Created `widgetIndexNum`